bliss/metrics.py

```diff
@@ -94,70 +107,14 @@ def compute(self) -> Dict[str, Tensor]:
-
-    total_n_matches: Tensor
-    total_coadd_n_matches: Tensor
-    total_coadd_gal_matches: Tensor
-    total_correct_class: Tensor
-    conf_matrix: Tensor
-    full_state_update: Optional[bool] = True
-
-    def __init__(
-        self,
-        slack: float = 1.0,
-        dist_sync_on_step: bool = False,
-    ) -> None:
-        """Computes matches between true and estimated locations.
-
-        Args:
-            slack: Threshold for matching objects a `slack` l-infinity distance away (in pixels).
-            dist_sync_on_step: See torchmetrics documentation.
-
-        Attributes:
-            total_n_matches: Total number of true matches.
-            total_correct_class: Total number of correct classifications.
-            Confusion matrix: Confusion matrix of galaxy vs. star
-        """
-        super().__init__(dist_sync_on_step=dist_sync_on_step)
-
-        self.slack = slack
-
-        self.add_state("total_n_matches", default=torch.tensor(0), dist_reduce_fx="sum")
-        self.add_state("total_coadd_gal_matches", default=torch.tensor(0), dist_reduce_fx="sum")
-        self.add_state("total_correct_class", default=torch.tensor(0), dist_reduce_fx="sum")
-        self.add_state("conf_matrix", default=torch.tensor([[0, 0], [0, 0]]), dist_reduce_fx="sum")
-
-    # pylint: disable=no-member
-    def update(self, true, est):
-        """Update the internal state of the metric including correct # of classifications."""
-        assert true.batch_size == est.batch_size
-        for b in range(true.batch_size):
-            ntrue, nest = true.n_sources[b].int().item(), est.n_sources[b].int().item()
-            tlocs, elocs = true.plocs[b], est.plocs[b]
-            tgbool, egbool = true["galaxy_bools"][b].reshape(-1), est["galaxy_bools"][b].reshape(-1)
-            if ntrue > 0 and nest > 0:
-                mtrue, mest, dkeep, _ = match_by_locs(tlocs, elocs, self.slack)
-                tgbool = tgbool[mtrue][dkeep].reshape(-1)
-                egbool = egbool[mest][dkeep].reshape(-1)
-                self.total_n_matches += len(egbool)
-                self.total_coadd_gal_matches += tgbool.sum().int().item()
-                self.total_correct_class += tgbool.eq(egbool).sum().int()
-                self.conf_matrix += confusion_matrix(tgbool, egbool, labels=[1, 0])
-
-    # pylint: disable=no-member
-    def compute(self) -> Dict[str, Tensor]:
-        """Calculate misclassification accuracy, and confusion matrix."""
-        return {
-            "n_matches": self.total_n_matches,
-            "n_matches_gal_coadd": self.total_coadd_gal_matches,
-            "class_acc": self.total_correct_class / self.total_n_matches,
-            "conf_matrix": self.conf_matrix,
+            "n_matches": self.gal_tp + self.gal_fp + self.gal_tn + self.gal_fn,
+            "n_matches_gal_coadd": self.gal_tp + self.gal_fn,
+            "class_acc": (self.gal_tp + self.gal_tn)
+            / (self.gal_tp + self.gal_fp + self.gal_tn + self.gal_fn),
```

Comment:
Tough to parse. Could you define the part of this calculation, e.g., the denominator immediately before the return statement?

Reply:
Sure!

Reply:
Yep, thanks, that's what I had in mind